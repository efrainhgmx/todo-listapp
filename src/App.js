import React, { useState, useRef, useEffect } from 'react';
import { TodoList  } from './components/TodoList';
import { v4 as uuidv4 } from 'uuid';

const KEY = "todoApp.todos";

export const App = () => {
    const [todos, setTodos] = useState([]);
    const todoTaskRef = useRef();
    useEffect(() => {
        localStorage.setItem(KEY, JSON.stringify(todos))
    }, [todos]);

    useEffect(() => {
        const storedTodos = JSON.parse(localStorage.getItem(KEY));
        if (storedTodos) {
            setTodos(storedTodos);
        }
    }, []);

    const handleTodoAdd = () => {
        const task = todoTaskRef.current.value;

        if(task === '') return;

        setTodos((prevTodos) => {
            return[...prevTodos, {id: uuidv4(), task, completed: false}];
        })

        todoTaskRef.current.value = null;

    };

    const toggleTodo = (id) => {
        const newTodos = [...todos];
        const todo = newTodos.find((todo) => todo.id === id);
        todo.completed = !todo.completed;
        setTodos(newTodos);
    }

    const handleClearAll = () => {
        const newTodos = todos.filter((todo) => !todo.completed);
        setTodos(newTodos);
    }


    return (
        <main>
            <h1>Lista de Tareas</h1>
            <TodoList todos={ todos } toggleTodo={ toggleTodo }/>
            <input className="input-text" ref={ todoTaskRef } type="text" placeholder="Nueva tarea" />
            <button onClick={ handleTodoAdd }>📝</button>
            <button onClick={ handleClearAll }>🗑</button>
            <p>Te quedan { todos.filter((todo) => !todo.completed).length } tareas por hacer.</p>
        </main>
        
    )
}
