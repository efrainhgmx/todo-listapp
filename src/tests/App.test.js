import React  from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('Test on <App /> component', () => {
    test('Should be match with the snapshoot', () => {
      const wrapper = shallow( <App /> );
      
      expect( wrapper ).toMatchSnapshot();
    });
    
})
