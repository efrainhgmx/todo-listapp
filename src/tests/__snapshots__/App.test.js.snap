// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <App /> component Should be match with the snapshoot 1`] = `
<main>
  <h1>
    Lista de Tareas
  </h1>
  <TodoList
    todos={Array []}
    toggleTodo={[Function]}
  />
  <form
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      placeholder="Nueva tarea"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      📝
    </button>
    <button
      onClick={[Function]}
    >
      🗑
    </button>
  </form>
  <p>
    Te quedan 
    0
     tareas por hacer.
  </p>
  <p>
    (Para añadir una tarea escribela y da clíck en el botón verde, para borrar la tarea debes haberla completado marcando el cuadro blanco y después dar clíck en el botón rojo). 
  </p>
</main>
`;
